### DISCUSSION TITLE
Google oauth help?

### DISCUSSION THREAD

**Discussion by froyop12** [9p4/jellyfin-plugin-sso#106] [Q&A] [answered]
Hi, I'm having trouble with OID with google. Is there any documentation anywhere that can help me? Not really sure where to start. Thanks:)

**Comment by 9p4** [comment, marked as answer]
1. Go to the [Google Developers Console](https://console.cloud.google.com).
2. Create a project (such as "jellyfin")
3. Go to the "APIs & Services" section
4. Go to the "Credentials" subsection
5. Press "Create Credentials" then "OAuth Client ID"
6. Select "Web Application" and add your redirect URL to the authorized redirect URL section (ie "https://jellyfin.example.com/sso/OID/r/google")
7. Copy the Client ID and Secret after pressing "Create".
8. For the OpenID url, use "https://accounts.google.com"
9. Disable role checking and role claims

**Reply by froyop12** [reply]
Ok so I got to work once where the google page showed up. I clicked my name and then it said error processing request. It now says that everytime I try it.

**Reply by 9p4** [reply]
What does it say in the Jellyfin logs?

**Reply by froyop12** [reply]
[2023-02-05 11:22:19.031 -05:00] [INF] [32] Jellyfin.Api.Auth.CustomAuthenticationHandler: "CustomAuthentication" was not authenticated. Failure message: "Invalid token."

**Reply by froyop12** [reply]
[2023-02-05 19:55:22.246 -05:00] [ERR] [55] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL "GET" "/SSO/oid/p/google". There's a lot in here lol.

**Reply by 9p4** [reply]
can you upload the logs to a pastebin or a gist?

**Reply by froyop12** [reply]
https://gist.github.com/froyop12/f721d3e3e628f81379e8ac42c8d60900

Does this work?

**Reply by 9p4** [reply]
Yep. Can you also upload the XML configuration of the plugin?

**Reply by froyop12** [reply]
Sorry, do you know where that is on windows? I configured the plugin on the web ui and I'm not sure where to look.

**Reply by 9p4** [reply]
Should be in `%APPDATA%\jellyfin\config\plugins\configurations\SSO-Auth.xml`

**Reply by froyop12** [reply]
It won't let me attach it cause it's a DLL file, not an xml.

**Reply by 9p4** [reply]
Are you sure you're in the right directory? The DLL is different from the XML

**Reply by froyop12** [reply]
In app data, there is nothing for jellyfin for some reason... Don't know why

**Reply by 9p4** [reply]
Try to find it with the help of https://jellyfin.org/docs/general/administration/configuration/

**Reply by froyop12** [reply]
https://gist.github.com/froyop12/28ab6180ac3fdab74b88e152e2418eee

**Reply by 9p4** [reply]
`<Enabled>false</Enabled>` You don't have the provider enabled

**Reply by froyop12** [reply]
Still doesn't like it. "Error processing request"

**Reply by 9p4** [reply]
upload logs again

**Reply by froyop12** [reply]
https://gist.github.com/froyop12/9a720995eefd644a04a511fe09b5a947

**Reply by froyop12** [reply]
Do I need an API key from google?

**Reply by 9p4** [reply]
no that's already in there

**Reply by 9p4** [reply]
Can you set the role claim configuration to anything (like "aaaa")

**Reply by froyop12** [reply]
Still the same error.

**Reply by 9p4** [reply]
anything different in the logs?

**Reply by froyop12** [reply]
https://gist.github.com/froyop12/e600b12931c67f89c92078ad80d1324c

**Reply by 9p4** [reply]
Just to be sure, you're going to the link /sso/OID/p/google to start the authentication flow?

**Reply by froyop12** [reply]
WOAH. Ok  I was doing OID/r/google. NOT p. That works but it takes me to my login page. It does the sign in with google now.

**Reply by 9p4** [reply]
Does it work? Do you log in and get a numeric username?

**Reply by froyop12** [reply]
YES! IT WORKS. YOU ARE LITERALLY THE BEST. THANK YOU SO MUCH!

**Reply by 9p4** [reply]
Maybe it's time to revisit https://github.com/9p4/jellyfin-plugin-sso/issues/84

**Reply by rediculum** [reply]
@9p4 I followed your steps and I get a error 400 redirect_uri_mismatch from google.
![image](https://github.com/user-attachments/assets/6363d60f-274f-4416-a767-646be694b83f)
 Am I doing something wrong? I was trying 24h later with the same

**Comment by froyop12** [comment]
I guess so💀. Thank you for your help!

On Sun, Feb 5, 2023 at 8:57 PM 9p4 ***@***.***> wrote:

> Maybe it's time to revisit #84
> <https://github.com/9p4/jellyfin-plugin-sso/issues/84>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/9p4/jellyfin-plugin-sso/discussions/106#discussioncomment-4877885>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMDHIYH3UPHWT4KXUVCN3VDWWBLAVANCNFSM6AAAAAAUR75AVU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 

All the best,

Ethan Parillo
